src/haystack/val/coord.rs

```diff
@@ -71,20 +71,23 @@ impl Eq for Coord {}
 
 impl PartialOrd for Coord {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(
-            self.lat
-                .to_bits()
-                .cmp(&other.lat.to_bits())
-                .cmp(&self.long.to_bits().cmp(&other.long.to_bits())),
-        )
+        Some(self.cmp(other))
     }
 }
 
 impl Ord for Coord {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.lat
-            .to_bits()
-            .cmp(&other.lat.to_bits())
-            .cmp(&self.long.to_bits().cmp(&other.long.to_bits()))
+        // Order by latitude first, then longitude
+        if self.lat < other.lat {
```

Comment:
Since `lat` and `long` are `f64`, a partial comparison is made, which is kind of odd within the `Ord` implementation.

Reply:
I change this to use partial_ord so it covers the NaN case.